vpdq/cpp/regtest.py

```diff
@@ -72,91 +74,114 @@ def get_argparse() -> argparse.ArgumentParser:
+    verbose = args.verbose
+
+    if not inputVideoFolder.exists():
+        raise argparse.ArgumentTypeError(
+            f"inputVideoFolder: {inputVideoFolder} does not exist."
+        )
+
+    with TemporaryDirectory() as tempOutputHashFolder:
+        """
+        Write the files to temp
+        If outputHashFolder is specified then
+        copy the output files there, overwriting existing files
+        """
+
+        tempOutputHashFolder = Path(tempOutputHashFolder)
+        if outputHashFolder is not None:
+            if not outputHashFolder.exists():
+                print(f"Creating output directory at {outputHashFolder}")
+                outputHashFolder.mkdir(parents=True, exist_ok=True)
+            print(f"Writing output to directory: {outputHashFolder}")
+        else:
+            print(f"Writing output to temp directory: {tempOutputHashFolder}")
+        # TODO: Add more general options for other video extensions.
+        for fileStr in glob.iglob(f"{inputVideoFolder}/**/*.mp4", recursive=True):
+            file = Path(fileStr)
+
+            """
+            Create output hash file or overwrite existing file
+            This is hardcoded in cpp, and it
+            does not create the file if it does not exist:
+
+            vpdq-hash-video.cpp:
+
+            // Strip containing directory:
+            std::string b = basename(inputVideoFileName, "/");
+            // Strip file extension:
+            b = stripExtension(b, ".");
+            outputHashFileName = outputDirectory + "/" + b + ".txt";
+            """
+            outputHashFile = tempOutputHashFolder / f"{file.stem}.txt"
+            with open(outputHashFile, "x+t"):
+                pass
+            print(f"\nHashing file {file.name}")
+            command = [
+                hashVideoExecutable,
+                "-r",
+                secondsPerHash,
+                "-d",
+                tempOutputHashFolder,
+                "-s",
+                downsampleFrameDimension,
+                "-i",
+                file,
+            ]
+
             if verbose:
-                subprocess.call(
-                    [
-                        "./build/match-hashes-byline",
-                        "-v",
-                        sampleFile,
-                        outputFile,
-                        distanceTolerance,
-                        qualityTolerance,
-                    ]
-                )
-            else:
-                subprocess.call(
-                    [
-                        "./build/match-hashes-byline",
-                        sampleFile,
-                        outputFile,
-                        distanceTolerance,
-                        qualityTolerance,
-                    ]
-                )
+                command.insert(1, "-v")
+
+            try:
+                subprocess.check_call(command)
+            except subprocess.CalledProcessError as e:
+                print(e)
+                sys.exit(1)
```

Comment:
blocking:  I think this is the same as not catching the exception (that check_call throws exception is well known). Am I missing something?

Reply:
Yeah looking at it this is stupid. I think I was looking at `vpdq-release.py` as a model, but that is definitely NOT the file I want to use as a reference.

Also check_call is old and replaced by check=True in subprocess.run(). I'll fix this.